device/common/include/traccc/clusterization/device/impl/reduce_problem_cell.ipp

```diff
@@ -48,15 +39,16 @@ inline void reduce_problem_cell(
          * impossible for that cell to ever be adjacent to this one.
          * This is a small optimisation.
          */
-        if (cells[j].channel1 + 1 < c1 || cells[j].module_link != mod_id) {
```

Comment:
I wonder if it is due to the fact that this if statement and the one below are no longer using the in-register information about the channels, but rather fetching it again from memory. :thinking:

Reply:
:thinking: But all of these functions are declared `inline`. And on these specific calls we were accessing the `channel1` and `module_link` variables directly from global memory to begin with.

But I also don't have better ideas. We'll need to revert the changes one-by-one, and see what makes a difference.

Reply:
At the same time, I was hoping that running a profiling on an Intel GPU may reveal something. I don't know, maybe VTune would have a more intuitive interface. :thinking: But at least on my laptop's iGPU I didn't find anything really interesting...

![image](https://github.com/acts-project/traccc/assets/30694331/037fc573-25b3-46fd-aa1a-7ea8980da5af)

Unlike on our NVIDIA GPUs, the `ccl_kernel` is not even the slowest kernel on this hardware... :confused: